controllers/subscriptions.go

```diff
@@ -259,9 +262,52 @@ func GetVendorCsvNames(cli client.Client, kind odfv1alpha1.StorageKind) ([]strin
 		csvNames = []string{OcsSubscriptionStartingCSV, RookSubscriptionStartingCSV, NoobaaSubscriptionStartingCSV,
 			PrometheusSubscriptionStartingCSV, RecipeSubscriptionStartingCSV}
 
-		if isProvider, err = isProviderMode(cli); !isProvider {
+		isProvider, err = isProviderMode(cli)
+		if err != nil {
+			return csvNames, err
+		}
+
+		if !isProvider {
 			csvNames = append(csvNames, OcsClientSubscriptionStartingCSV, CSIAddonsSubscriptionStartingCSV)
 		}
+
+		// In provider mode, upgrades of the ocs-client-operator and csiaddons are managed by the provider, not the odf-operator.
+		// This can result in these operators lagging behind the odf-operator, with different CSV versions.
+		// Therefore, we need to fetch the CSV name from the operator deployment.
+		// We are only fetching the ocs-client-operator CSV name and ignoring the csiaddons
+		// Because the ocs-client-operator is essential when enabling provider mode as we need to bring it up.
+
+		// Fetch the CSV name from the client operator deployment and append it to the csv list
+		if isProvider {
+			// get ocs-client-operator deployment with label
+			deployments := &appsv1.DeploymentList{}
+			err = cli.List(context.TODO(), deployments, &client.ListOptions{
+				Namespace:     OperatorNamespace,
+				LabelSelector: labels.SelectorFromSet(map[string]string{"app": "ocs-client-operator"}),
+			})
+
+			if err != nil {
+				return csvNames, err
+			}
+
+			if len(deployments.Items) == 0 {
+				return csvNames, fmt.Errorf("ocs-client-operator deployment not found")
+			}
+
+			// get owner ref index from deployment
+			ownerRefIndex := slices.IndexFunc(deployments.Items[0].OwnerReferences, func(o metav1.OwnerReference) bool {
+				return o.Kind == "ClusterServiceVersion"
+			})
+			if ownerRefIndex == -1 {
+				return csvNames, fmt.Errorf("ClusterServiceVersion owner reference not found in ocs-client-operator deployment")
+			}
+
+			ownerRef := deployments.Items[0].OwnerReferences[ownerRefIndex]
+
+			// get csv name from owner ref
+			csvNames = append(csvNames, ownerRef.Name)
+			OcsClientSubscriptionStartingCSV = ownerRef.Name
```

Comment:
You are inconsistent with the naming convention. Both OCS and CSV are abbreviations but, you use camel-case for OCS and use uppercase for CSV. In most new operators we use camel-case for everything to prevent the issue of having two or more abbreviations one after the other, which in the case of the upper case makes the variable/field/func name unreadable.  

Regardless of which letter-casing you decide to use, the request is to stay consistent

Reply:
`OcsClientSubscriptionStartingCSV` is a global variable which is already defined in the package.